src/main/java/Core/MessageListener.java

```diff
@@ -0,0 +1,35 @@
+package Core;
+
+import java.io.*;
+
+public class MessageListener implements Runnable {
+
+    private BufferedReader reader;
+    private PrintStream output;
+    private boolean stop;
+
+    public MessageListener(InputStream inputStream, PrintStream output) {
```

Comment:
I think this class is pretty well setup 👍 
One thing I would consider is the fact that `InputStream` and `PrintStream` are at different levels of abstraction, 
which can spot by the fact that you can use the `output` ASAP while you still need to bubble-wrap the `inputStream` in a lovely InputStreamReader and BufferedReader before using it.